code/game/machinery/tesla_turret.dm

```diff
@@ -479,8 +482,16 @@ GLOBAL_LIST_INIT(turret_channels, new/list(5))
 	var/list/targets = list()			//list of primary targets
 	var/list/secondarytargets = list()	//targets that are least important
 
-	for(var/mob/M in mobs_in_view(world.view, src))	// Why do we check every mob in an 15x15 area (Or radius of 7) if we later limit it to a radius of 6 (or area of 13x13) -R4d6
-		assess_and_assign(M, targets, secondarytargets)
+	for(var/mob/living/M in view(firing_range, src)) //WE USED WORLD.VIEW BEFORE THATS FUCKING PSYCHOPATHIC
+		assess_and_assign(M, targets, secondarytargets) //might want to not use a proc due to proc overhead cost
+
+	for(var/atom/A in GLOB.mechas_list)
+		if (A.z == z) //z-level check
+			if((get_dist(A, src) < firing_range) && can_see(src, A, firing_range))
+				var/obj/mecha/mech = A
+				var/mob/living/occupant = mech.get_mob()
+				if (occupant)
+					assess_and_assign(occupant, targets, secondarytargets)
```

Comment:
```suggestion
	for(var/i all_mobs_in_view(firing_range, src)
		assess_and_assign(i, targets, secondarytargets) //might want to not use a proc due to proc overhead cost
```

Reply:
all_mobs_in_view is less performant, the algorithm here is faster for the same effect